tests/unit/test_unit_smhi.py

```diff
@@ -0,0 +1,249 @@
+"""SMHI unit tests."""
+
+from unittest.mock import MagicMock, patch
+
+import pandas as pd
+import pytest
+from smhi.smhi import SMHI
+
+
+class TestUnitSMHI:
+    """Unit tests for SMHI class."""
+
+    @patch("smhi.metobs.Parameters.__new__")
+    def test_unit_smhi_init(self, mock_parameters):
+        """Unit test for SMHI init method.
+
+        Args:
+            mock_requests_metobs: mock requests metobs object
+        """
+        client = SMHI()
+
+        assert client.parameters
```

Comment:
You aren't asserting anything here if `client.parameters` doesn't return a `bool`, which I think it doesn't. You need to compare it to something.

Reply:
It would have failed if client.parameters wasn't there.

Reply:
Sure, makes sense that something not None/False returned will pass the test, but I think a clearer test would be to assert what that returned value should be

```python
assert client.parameters == mock_parameters.return_value
```

Reply:
Yes I tried that but it didn't work :) I can give it a go again once I've implemented that class you suggested, will work a few hours to set this straight this morning.